elmslie-android/src/main/java/vivid/money/elmslie/android/base/ElmFragment.kt

```diff
@@ -16,12 +17,25 @@ abstract class ElmFragment<Event : Any, Effect : Any, State : Any> : Fragment,
     constructor(@LayoutRes contentLayoutId: Int) : super(contentLayoutId)
 
     @Suppress("LeakingThis", "UnusedPrivateMember")
-    private val elm = ElmScreen(this, lifecycle) { requireActivity() }
+    private val elm = ElmScreen(this, lifecycle)
 
     protected val store
         get() = storeHolder.store
 
     override val storeHolder: StoreHolder<Event, Effect, State> by fastLazy {
         LifecycleAwareStoreHolder(lifecycle, ::createStore)
     }
-}
\ No newline at end of file
+
+    final override val shouldStopOnProcessDeath: Boolean = hasStopElmOnProcessDeathInHierarchy
+}
+
+val Fragment.hasStopElmOnProcessDeathInHierarchy: Boolean
```

Comment:
Could you explain the reasoning behind the decision to traverse fragment hierarchy? it's not very clear to me. The same thing could be implemented by marking all fragments with StopElmOnProcessDeath interface, but with this change there will be no way to opt out of this for a child fragment

Reply:
Let's assume that we have MainFragment with 4 nested Fragments (tab bar). Inside MainFragment during restoring after process death we want to navigate to another Fragment (AuthFragment) and close all nested Fragments. So we need to stop not only MainFragment but nested Fragments too.

Reply:
Yes, that makes sense, but i look at it as a solution for all cases, not just handling logout. In general it has several downsides:
- Looks like nothing prevents annotating all child fragments with StopElmOnProcessDeath
- Child fragments could be reused without the parent fragment. Not annotating them at first might cause a bug, because the developer that will add this fragment as a root might not notice that the annotation is necessary
- Some chid fragments might not require stopping (in some rare future cases, for example, child fragment might handle logout) 
- If this logic is really necessary it can be implemented inside parent fragment on a per case basis
- In general, this is highly tied to the vivid.money implementation, but in other projects it might not be necessary

Reply:
And child Fragments can't know that MainFragments will be closed, so Init event will be called (that's we want to avoid)

Reply:
I assume that if child fragments wil be annotated, ElmStore would stop them the same way it stops MainFragment. What is the problem in this case?

Reply:
I'll explain my suggestion in case there's some misunderstanding. I think it would be better to isolate process death hanliding logic to a fragment level and allow each fragment to decide how to handle it. In case of nested fragments all fragments, parent and all children should be annotated.

Reply:
My intention was to avoid leaking knowledge about parent's logic to child fragments, but downsides you described look important. Maybe we need an even more generic mechanism to start\stop child fragment stores from parent fragment?

Something like:
```
  childFragmentManager.fragments.forEach { 
      (it as? ElmDelegate<*,*,*>)?.storeHolder?.store.stop()
  }
```

It will allow developers to stop stores explicitly, only when they are sure that it's needed (in our case when we will be sure that child fragments will not be reused and current fragment will be closed)

Reply:
I think this is a valid approach, but it has some problems:
- It won't handle cases with more layers of hieraechy, because the parent only stops direct children. Another thing is - it stops stores children directly and even explicit setting of shouldStopOnProcessDeath to false will not prevent it from stopping
- in this solution the logic of stopping elm is mixed up between Fragment and Elmstore
The first version, with traversing fragments, solved all them. I just don't think it should be the default one.

In general i suppose library users will have these popular cases for this feature
1. never stopping elm, for example, in apps without authorization
2. always stopping elm, in case all screens should be redirected to authoziation 
3. custom logic for specific cases, like the one you're desribing
4. some other cases, apart from authorization

The first two cases could be handled easily by adding a property to ElmslieConfig with false default value and an ability to override it. The fourth one is completely custom, so the ability to override ElmDelegate property should be enough. If there are cases for the third one, it would be better to add some helpers for handling them to the library. 

But if in your case just setting shouldStopOnProcessDeath to false would be enough it would be easier to postpone adding such helpers to library for now

Reply:
> 2. always stopping elm, in case all screens should be redirected to authoziation

With property inside ElmslieConfig all Fragments even AuthFragment will be stoped and we will not be able to use Elm inside it. We need to consider it if we want to add this property.

Reply:
Also, I understand that we don't need to stop ElmStore, we need to prevent it start. So here is an updated idea:
```
fun Fragment.traversingPreventElmToStart() {
   (this as? ElmDelegate<*,*,*>)?.preventToRunMvi = true
   childFragmentManager.fragments.forEach(::traversingPreventElmToStart)
}
```
We can add public property (or method) to prevent running MVI and give parents the ability to set it for all children.

Reply:
It would be enough to override the property explicitly to false in AuthFragment and all the other necessary cases